lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -12,10 +12,7 @@
   mk_resource_methods
 
   def exists?
-    inst_cmd = "Get-WebConfiguration -Filter '/system.applicationHost/applicationPools/add' | ? { $_.Name -eq '#{@resource[:name]}' }"
```

Comment:
`Notice: Applied catalog in 457.49 seconds` so the modified filter saves 120 seconds (down from 577 seconds) but still 200 seconds longer than the 257 seconds of no query.